forge/db/models/TeamBrokerClient.js

```diff
@@ -20,6 +20,26 @@ module.exports = {
     associations: function (M) {
         this.belongsTo(M.Team)
     },
+    hooks: function (M, app) {
+        return {
+            beforeCreate: async (project, opts) => {
+                // if the product is licensed, we permit overage
+                const isLicensed = app.license.active()
+                if (isLicensed !== true) {
+                    const { mqttClients } = await app.license.usage('mqttClients')
+                    if (mqttClients.count >= mqttClients.limit) {
+                        throw new Error('license limit reached')
+                    }
+                }
+            },
+            afterCreate: async (project, opts) => {
+                const { mqttClients } = await app.license.usage('mqttClients')
+                if (mqttClients.count > mqttClients.limit) {
+                    await app.auditLog.Platform.platform.license.overage('system', null, mqttClients)
+                }
+            }
+        }
+    },
```

Comment:
Do we need to generate a new license for prod (and staging) before this gets merged?

Reply:
Technically no, as we are licensed so allowed to go over the limits at the platform level - but we'll get audit log overage reports. So in practice, we'll want to deploy a new license around the same time as this merging - just doesn't have to be strictly in order.